src/app/calendar/parcelCalendarFunctions.ts

```diff
@@ -1,39 +1,32 @@
 import { Schema } from "@/database_utils";
-import supabase from "@/supabaseClient";
 import { CalendarEvent } from "@/components/Calendar/Calendar";
+import { RainbowPalette } from "styled-components";
 
-export type ParcelWithClientName = Schema["parcels"] & { clients: { full_name: string } | null };
+export type ParcelWithClientName = Schema["parcels"] & {
+    clients: { full_name: Schema["clients"]["full_name"] } | null;
+    collection_centres: { name: Schema["collection_centres"]["name"] } | null;
+};
 
 const COLLECTION_DURATION_MS = 30 * 60 * 1000;
 
 export interface LocationColorMap {
-    [location: string]: { color: string; text: string };
+    [location: Schema["collection_centres"]["name"] | "default"]: {
+        color: RainbowPalette[keyof RainbowPalette]["background"];
+        text: RainbowPalette[keyof RainbowPalette]["foreground"];
+    };
 }
 
-export const getParcelsWithCollectionDate = async (): Promise<ParcelWithClientName[]> => {
-    const { data, error } = await supabase
-        .from("parcels")
-        .select("*, clients ( full_name )")
-        .not("collection_datetime", "is", null);
-
-    if (error) {
-        throw new Error("Database error");
-    }
-
-    return data;
-};
-
 export const parcelsToCollectionEvents = (
     parcels: ParcelWithClientName[],
     colorMap: LocationColorMap
 ): CalendarEvent[] => {
     return parcels.map((parcel) => {
-        const fullName = parcel.clients?.full_name ?? "";
+        const fullName = parcel.clients!.full_name ?? "";
```

Comment:
you probably want `parcel.clients?.full_name` instead of `parcel.clients!.full_name`

Reply:
agree